Library/Homebrew/cmd/search.rb

```diff
@@ -89,15 +95,28 @@ def search
       remote_casks = remote_results[:casks]
       all_casks = local_casks + remote_casks
 
-      if all_formulae.any?
-        ohai "Formulae"
-        puts Formatter.columns(all_formulae)
-      end
+      if args.formulae? && !formulae_and_casks_present
+        if all_formulae.any?
+          ohai "Formulae"
+          puts Formatter.columns(all_formulae)
+        end
+      elsif args.casks? && !formulae_and_casks_present
+        if all_casks.any?
+          puts if all_formulae.any?
+          ohai "Casks"
+          puts Formatter.columns(all_casks)
+        end
+      else
+        if all_formulae.any?
+          ohai "Formulae"
+          puts Formatter.columns(all_formulae)
+        end
 
-      if all_casks.any?
-        puts if all_formulae.any?
-        ohai "Casks"
-        puts Formatter.columns(all_casks)
+        if all_casks.any?
+          puts if all_formulae.any?
+          ohai "Casks"
+          puts Formatter.columns(all_casks)
+        end
       end
```

Comment:
```suggestion
```

Reply:
Thank you for the feedback @MikeMcQuaid! I understand the reason behind all these changes, but I'd like to point out that the following case is being missed:
`brew search <argument>`

I think that we'll have to add another check for cases where no flags are passed, and display both casks and formulae in such cases.

Reply:
Good point!